### PR TITLE
Remove `gfx_finish` config variable, remove `dbg_hitch` config variable, improve config variable descriptions, add missing descriptions

### PR DESCRIPTION
**Pull request by Robyt3**
## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Review by edg-l — COMMENTED** [review]
better comment

File: src/engine/shared/config_variables.h [hunk 1]
```diff
@@ -121,12 +121,12 @@ MACRO_CONFIG_INT(GfxColorDepth, gfx_color_depth, 24, 16, 24, CFGFLAG_SAVE | CFGF
 MACRO_CONFIG_INT(GfxVsync, gfx_vsync, 0, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Vertical sync (may cause delay)")
 MACRO_CONFIG_INT(GfxDisplayAllVideoModes, gfx_display_all_video_modes, 0, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Show all video modes")
 MACRO_CONFIG_INT(GfxHighDetail, gfx_high_detail, 1, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "High detail")
-MACRO_CONFIG_INT(GfxFsaaSamples, gfx_fsaa_samples, 0, 0, 64, CFGFLAG_SAVE | CFGFLAG_CLIENT, "FSAA Samples")
+MACRO_CONFIG_INT(GfxFsaaSamples, gfx_fsaa_samples, 0, 0, 64, CFGFLAG_SAVE | CFGFLAG_CLIENT, "FSAA samples (may cause delay)")
 MACRO_CONFIG_INT(GfxRefreshRate, gfx_refresh_rate, 0, 0, 10000, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Screen refresh rate")
-MACRO_CONFIG_INT(GfxFinish, gfx_finish, 0, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "")
+MACRO_CONFIG_INT(GfxFinish, gfx_finish, 0, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Finish rendering before update")
```
Comment (edg-l) [inline src/engine/shared/config_variables.h:126]: "Wait for the GPU to finish the current execution before doing a new cycle"

Comment (Jupeyy) [reply src/engine/shared/config_variables.h:126]: But only if gfx_refresh_rate is 0 xdddd

else check if refresh cycle is hit, then 100% wait for the GPU, even if the GPU is not finished executing (=> causes a wait)

File: src/engine/shared/config_variables.h [hunk 2]
```diff
@@ -121,12 +121,12 @@ MACRO_CONFIG_INT(GfxColorDepth, gfx_color_depth, 24, 16, 24, CFGFLAG_SAVE | CFGF
...
 MACRO_CONFIG_INT(GfxBackgroundRender, gfx_backgroundrender, 1, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Render graphics when window is in background")
 MACRO_CONFIG_INT(GfxTextOverlay, gfx_text_overlay, 10, 1, 100, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Stop rendering textoverlay in editor or with entities: high value = less details = more speed")
-MACRO_CONFIG_INT(GfxAsyncRenderOld, gfx_asyncrender_old, 1, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Do rendering async from the the update")
+MACRO_CONFIG_INT(GfxAsyncRenderOld, gfx_asyncrender_old, 1, 0, 1, CFGFLAG_SAVE | CFGFLAG_CLIENT, "Do updates independent from rendering and refresh rate")
```
[the remark this answers is not in the document]
Comment (Jupeyy) [reply src/engine/shared/config_variables.h:129]: But only if gfx_refresh_rate is 0 xdddd

else check if refresh cycle is hit, then 100% wait for the GPU, even if the GPU is not finished executing (=> causes a wait)

[the remark this answers is not in the document]
Comment (Jupeyy) [reply src/engine/shared/config_variables.h:129]: We need the opposite of this:
"During a render cycle, wait for the previous render cycle to finish"

File: src/engine/shared/config_variables.h (hunk 1 above)
Comment (Jupeyy) [reply src/engine/shared/config_variables.h:126]: I don't even see the sense of this config at all tbh

Why would someone want to finish the current frame before swapping the content of it

The GPU driver does this anyway.. else u'd see broken frames quite often :D (double+ buffering)

Comment (Robyt3) [reply src/engine/shared/config_variables.h]: Removed `gfx_finish`.

File: src/engine/shared/config_variables.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Robyt3) [reply src/engine/shared/config_variables.h]: It's now "During a render cycle, proceed without waiting for the previous render cycle to finish"

[the remark this answers is not in the document]
Comment (Jupeyy) [reply src/engine/shared/config_variables.h]: that sounds like a perfect description for VSync xDDD

[the remark this answers is not in the document]
Comment (Jupeyy) [reply src/engine/shared/config_variables.h]: Yes and no xd

"During a non rendering-related cycle, proceed without doing a render cycle, if the render cycle would need to wait for the previous render cycle to finish"

I think that would be pretty correct xDD

[the remark this answers is not in the document]
Comment (Jupeyy) [reply src/engine/shared/config_variables.h]: Assuming that update cycle is the superset of all possible cycles (network etc.)

Maybe we shouldn't do a religion out of this stupid description that nobody will ever read anyway xD

And even if someone reads it.. he will ask in dc anyway xDD

**Pull request merged**